### PR TITLE
Javascript2 week3/niloufar

### PR DESCRIPTION
**Pull request by Niloufar97**
Hi, I'm done with this part. I asked you how to separate flipping-back cards and matching them. I spent more than one day working on it and I watched a lot of solutions on YouTube and websites. but it makes me confused ( I also created a branch to practice that but it didn't work). 
At the end, I added both functionality and I found it easier and more understandable.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -64,27 +82,124 @@ const createCardDiv = (card) => {
...
+  let second = 0;
+  let minute = 0;
+  timer = setInterval(() => {
+    second++;
+    if (second >= 60) {
+      second = 0;
+      minute++;
+    }
+    seconds.textContent = second;
+    minutes.textContent = minute;
```
Comment (shpomp) [inline script.js]: It would be more conventional and look nicer if the time was displayed in a mm:ss format 🙏🏽

File: script.js [hunk 2]
```diff
@@ -64,27 +82,124 @@ const createCardDiv = (card) => {
...
       });
     });
   };
   renderShuffledCards();
 };
 updateCards();
+
+// click handler function----------------------------------------------------
+
+function clickHandker(cardDiv, card) {
```
Comment (shpomp) [inline script.js]:
```suggestion
function clickHandler(cardDiv, card) {
```

**Review by shpomp — APPROVED** [review]
Looks good, @Niloufar97 ! 
And you are going a little bit ahead with the functionality 👏🏽 

I think I understand what you mean with flipping the card down - if it flips back automatically, then you don't need the functionality to flip them down manually. 

I don't have many remarks, you are doing a very good job! 
If you want to try to improve things, you could look into a few aspects, optionally:
- I am not sure that you need both movementsCounter and clickCounter. They serve a very similar purpose, so maybe you could go on with 1 counter and refactor accordingly. 
- You could look into an approach where the pictures array objects will already have an ID, as that is how it would normally look like coming from an API. Then your code would; also have to change a bit. 

That is it. Keep up the nice progress! 👍🏽 ⭐

File: script.js (hunk 1 above)
Comment (Niloufar97) [reply script.js]: done :)

**Comment by Niloufar97** [comment]
> * I am not sure that you need both movementsCounter and clickCounter. They serve a very similar purpose, so maybe you could go on with 1 counter and refactor accordingly.

Hi Karo, merry Christmas and happy new year. thanks for reviewing my homework.

I tried to use only clickCounter like this:
  if (selectedCards.length === 2) {
    checkMatch();
    movementsContainer.textContent = clickCounter/2;
  }
but if I click on one card twice, I will get silly results ( for example 3.5). I also tried this:
  if(clickCounter%2 === 0){
    movementsContainer.textContent = clickCounter/2;
  }
But it has the same problem if I click on one card twice it counts as one movement but in fact, it is not :)
I think having two counters is more accurate. or maybe I'm doing something wrong. I would appreciate it if you could show me the best way of doing that.

thanks

**Comment by shpomp** [comment]
> > * I am not sure that you need both movementsCounter and clickCounter. They serve a very similar purpose, so maybe you could go on with 1 counter and refactor accordingly.
> 
> Hi Karo, merry Christmas and happy new year. thanks for reviewing my homework.
> 
> I tried to use only clickCounter like this: if (selectedCards.length === 2) { checkMatch(); movementsContainer.textContent = clickCounter/2; } but if I click on one card twice, I will get silly results ( for example 3.5). I also tried this: if(clickCounter%2 === 0){ movementsContainer.textContent = clickCounter/2; } But it has the same problem if I click on one card twice it counts as one movement but in fact, it is not :) I think having two counters is more accurate. or maybe I'm doing something wrong. I would appreciate it if you could show me the best way of doing that.
> 
> thanks

@Niloufar97  sorry, I did not see this before :/
Just missed the notification. 
If you still want to talk about this, just ping me on Slack!

**Pull request merged**